pkg/build/pipelines/git-checkout.yaml

```diff
@@ -1,148 +1,181 @@
+                return 0
+            fi
+            if [ ! -f "$cpicksf" ]; then
+                msg "cherry picks input '$cpicksf' is not a file"
+                return 1
+            fi
+
+            local line="" branch="" hash=""
+            while read line; do
+                # drop anything after #
+                line=${line%%#*}
+                [ -z "$line" ] && continue
+                case "$line" in
+                    */*) branch=${line%%/*}; hash=${line#*/};;
+                    *) hash=${line};;
+                esac
+                if [ -n "$branch" ]; then
+                    case " $fetched_branches " in
+                        *) vr git fetch origin $branch:$branch || {
+                            msg "failed to fetch branch $branch"
+                            return 1
+                            }
+                            local fetched_branches="$fetched_branches $branch";;
+                    esac
+                fi
+                vr env GIT_AUTHOR_NAME="Melange Build" GIT_AUTHOR_EMAIL="melange-build@cgr.dev" GIT_COMMITTER_DATE="$SOURCE_EPOCH_DATE" git cherry-pick -x "$hash" || {
+                    msg "failed to cherry-pick $hash from branch $branch"
+                    return 1
+                }
+                count=$((count + 1))
+            done < "$cpicksf"
+            msg "applied $count cherry-pick. head is now $(git rev-parse HEAD)"
+            rm /tmp/cherry-pick
+        }
+
+        main() {
+            local repo=$1 dest=${2:-.} depth=${3:-"-1"} branch=$4
+            local tag=$5 expcommit=$6 recurse=${7:-false}
+            local cherry_pick="$8"
+            msg "repo='$repo' dest='$dest' depth='$depth' branch='$branch'" \
+                "tag='$tag' expcommit='$expcommit' recurse='$recurse'"
+
+            case "$recurse" in
+                true|false) :;;
+                *) fail "recurse must be true or false, not '$recurse'"
+            esac
+
+            [ -n "$repo" ] || fail "repository not provided"
+
+            if [ -z "$branch" ] && [ -z "$tag" ]; then
+                msg "Warning: you have not specified a branch or tag."
+            fi
+
+            [ -n "$expcommit" ] ||
+                msg "Warning: no expected-commit"
+
+            local flags="" depthflag="" dest_fullpath="" workdir=""
+            local remote="origin" rcfile="" rc=""
+            [ -n "$branch" ] && flags="--branch=$branch"
+            [ -n "$tag" ] && flags="--branch=$tag"
+            [ "$recurse" = "true" ] && flags="$flags --recurse-submodules"
+
+            [ "$depth" = "-1" ] || depthflag="--depth=$depth"
+
+            workdir=$(mktemp -d)
+            rcfile=$(mktemp)
+            mkdir -p "$dest"
+            dest_fullpath=$(realpath "$dest")
+
+            vr git config --global --add safe.directory "$workdir"
+            vr git config --global --add safe.directory "$dest_fullpath"
+
+            vr git clone "--origin=$remote" $flags \
```

Comment:
can you add `--config=advice.detachedHead=false` here ?
that will silence that big git detached head message.